lib/attach/model_extension.rb

```diff
@@ -141,7 +141,11 @@ def attachment(name, options = {}, &block)
 
         define_method "#{name}=" do |file|
           if file.is_a?(Attach::Attachment)
+            self.try(name)&.destroy
```

Comment:
Not sure this should happen when setting the attribute. It needs to stay in the database until the new file has been added and verified. There is an array of attachments which should be removed you can add it to I think.